Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -646,6 +647,211 @@ void orient_to_bound_a_volume(TriangleMesh& tm)
+ *                     If `PolygonMesh` has an internal property map for `CGAL::face_index_t`,
+ *                     as a named parameter, then it must be initialized.
+ * @tparam NamedParameters a sequence of \ref pmp_namedparameters
+ *
+ * @param pm a surface mesh
+ * @param np optional sequence of \ref pmp_namedparameters among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{vertex_point_map}
+ *     the property map with the points associated to the vertices of `tm`.
+ *     If this parameter is omitted, an internal property map for
+ *     `CGAL::vertex_point_t` must be available in `PolygonMesh`
+ *   \cgalParamEnd
+ *   \cgalParamBegin{face_index_map}
+ *     a property map containing an index for each face initialized from 0 to num_faces(pm).
+ *   \cgalParamEnd
+ *   \cgalParamBegin{maximum_number_of_faces}
+ *     if not 0 (default), a connected component is considered reversible only
+ *     if it has not more faces than the value given. Else, it is always considered reversible.
+ *   \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template <class PolygonMesh, class NamedParameters>
+void merge_reversible_connected_components(PolygonMesh& pm,
+                                           const NamedParameters& np)
+{
+  typedef boost::graph_traits<PolygonMesh> GrT;
+  typedef typename GrT::face_descriptor face_descriptor;
+  typedef typename GrT::halfedge_descriptor halfedge_descriptor;
+
+  typedef typename 
+      GetVertexPointMap<PolygonMesh, NamedParameters>::const_type Vpm;
+
+  typedef typename boost::property_traits<Vpm>::value_type Point_3;
+  Vpm vpm = parameters::choose_parameter(parameters::get_parameter(np, internal_np::vertex_point),
+                                get_const_property_map(vertex_point, pm));
+
+  typedef std::size_t F_cc_id;
+  typedef std::size_t B_cc_id;
+
+  typedef typename
+      GetFaceIndexMap<PolygonMesh, NamedParameters>::type Fidmap;
+
+  Fidmap fim = parameters::choose_parameter(parameters::get_parameter(np, internal_np::face_index),
+                                   get_const_property_map(face_index, pm));
+
+  typedef dynamic_face_property_t<F_cc_id>                   Face_property_tag;
+  typedef typename boost::property_map<PolygonMesh, Face_property_tag>::type   Face_cc_map;
+  Face_cc_map f_cc_ids  = get(Face_property_tag(), pm);
+  F_cc_id nb_cc = connected_components(pm, f_cc_ids, parameters::face_index_map(fim));
+
+  std::vector<std::size_t> nb_faces_per_cc(nb_cc, 0);
+  for (face_descriptor f : faces(pm))
+    nb_faces_per_cc[ get(f_cc_ids, f) ]+=1;
+
+  std::map< std::pair<Point_3, Point_3>, std::vector<halfedge_descriptor> > border_hedges_map;
+  std::vector<halfedge_descriptor> border_hedges;
+  typedef typename boost::property_map<PolygonMesh, dynamic_halfedge_property_t<B_cc_id> >::type H_to_bcc_id;
+  H_to_bcc_id h_bcc_ids = get(dynamic_halfedge_property_t<B_cc_id>(), pm);
+  const B_cc_id DV(-1);
```

Comment:
Would definitely be better if the variable had a comprehensible name, it's an algo sufficiently difficult to follow